pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -642,6 +655,24 @@ async def _completions_create(
                 extra_body=model_settings.get('extra_body'),
             )
         except APIStatusError as e:
+            if azure_error := _check_azure_content_filter(e, self.system):
+                resp = chat.ChatCompletion(
+                    id='content_filter',
+                    choices=[
+                        chat.chat_completion.Choice(
+                            finish_reason='content_filter',
+                            index=0,
+                            message=chat.ChatCompletionMessage(content=None, role='assistant'),
+                        )
+                    ],
+                    created=0,
+                    model=self.model_name,
+                    object='chat.completion',
+                )
+                # Attach inner error details for _process_provider_details to pick up
+                if inner_error := azure_error.get('innererror'):
+                    setattr(resp, 'azure_content_filter_result', inner_error)
```

Comment:
Hmm, now that we're having to resort to this, I do think it'd be easier to have `_check_azure_content_filter` return a `ModelResponse` directly, and then change the signature (and call sites) of this method to allow that as an alternative to `chat.ChatCompletion`.

That'd be similar to what we do here:

https://github.com/pydantic/pydantic-ai/blob/c1213e4a7a978099b64c642b38e1e041d18afad6/pydantic_ai_slim/pydantic_ai/models/groq.py#L202-L221